core/arch/arm/include/arm64_macros.S

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2015, Linaro Limited
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+	.altmacro
+
+	.macro __do_dregs op1, op2, reg_bytes, base_reg, base_offs, reg0, reg1
+		.if \reg_bytes == 8
+			\op1\op2	x\reg0, x\reg1, [\base_reg, #\base_offs]
+		.else
+			\op1\op2	w\reg0, w\reg1, [\base_reg, #\base_offs]
+		.endif
+	.endm
+	.macro __do_reg op1, op2, reg_bytes, base_reg, base_offs, reg
+		.if \reg_bytes == 8
+			\op1\op2	x\reg, [\base_reg, #\base_offs]
+		.else
+			\op1\op2	w\reg, [\base_reg, #\base_offs]
+		.endif
+	.endm
+	.macro _do_regs op, reg_bytes, base_reg, base_offs, from_regnum, \
```

Comment:
op is coming the instruction (st/ld) and in the subsequent calls it's passed/labelled as op1. I think it's a bit hard to follow this. Could we simply rename op and op1 to "ins" or something else that differentiate it a bit and gives a hint that it's a "prefix" of an assembler instruction?